### ISSUE TITLE
LastFM explicit track scrobble issue

### ISSUE THREAD

**Issue by anameshouldntbeabigdeal** [Inrixia/neptune-plugins#83]
## Issue

Sometimes when scrobbling an explicit song, it's scrobbled with the explicit tag in the title on LastFM.

### Example

Song Link: https://tidal.com/browse/track/258373418?u

How it's scrobbled on LastFM

![image](https://github.com/user-attachments/assets/34f726e1-dc95-4fdf-8174-d100668aa546)

This isn't the case for every explicit song however.

Song Link: https://tidal.com/browse/track/387049122?u

How it's scrobbled

![image](https://github.com/user-attachments/assets/051ce16e-9a5f-49da-9ba3-db19031a0c1a)

I'm not sure why exactly this is the case for some songs over others.

### Other examples with the issue

1 - https://tidal.com/browse/track/320951442?u
2 - https://tidal.com/browse/track/345556046?u
3 - https://tidal.com/browse/track/77706588?u
4 - https://tidal.com/browse/track/332202523?u

### Other examples without the issue

1 - https://tidal.com/browse/track/388395180?u
2 - https://tidal.com/browse/track/377661774?u
3 - https://tidal.com/browse/track/385364814?u
4 - https://tidal.com/browse/track/387094039?u

**Comment by Inrixia** [comment]
Just leaving a comment here as I realize I haven't.

This is a issue with Tidal not being great at maintaining their track Metadata.

Thankfully all the examples you have shared should be easy to filter for by not adding a tracks disambiguation if it's a specific word, ie "explicit".

I'll get around to adding this in when I have the chance. Thanks for reporting it! ❤️

**Comment by Alebacce** [comment]
Here I have some messy metadata examples from Tidal.

https://tidal.com/browse/track/68639028?u
<img width="622" alt="Screenshot 2024-09-23 alle 15 22 31" src="https://github.com/user-attachments/assets/943e30e0-370c-49ab-832d-261b27fea5cf">
Here it appends in a wrong way (48kHz/24bit)

https://tidal.com/browse/track/168110907?u
<img width="622" alt="Screenshot 2024-09-23 alle 15 23 26" src="https://github.com/user-attachments/assets/d260cbba-60b9-4c1a-a6c7-5dab07c9dbdf">
Here (explicit), it should be (deluxe version)

https://tidal.com/browse/track/68639135?u
<img width="622" alt="Screenshot 2024-09-23 alle 15 23 48" src="https://github.com/user-attachments/assets/91e67bd9-e216-4d7e-a91f-39c82f2f0d95">
Here (Mastered For iTunes)

https://tidal.com/browse/track/68639002?u
<img width="622" alt="Screenshot 2024-09-23 alle 15 24 28" src="https://github.com/user-attachments/assets/5d2acddf-ff52-4b62-a6ac-be28da91badf">
Here (24-bit 48kHz)

https://tidal.com/browse/track/152246358?u
<img width="622" alt="Screenshot 2024-09-23 alle 15 24 59" src="https://github.com/user-attachments/assets/5c79607a-fb1a-40ce-afc7-4a8e91556c67">
Here it changes the album name entirely, it goes from Folklore (deluxe version) to Foikiore (deluxe9, what a twist

In the list below screens already contains original right version of the album name. Below every input there is the wrong version recived by last.fm
https://tidal.com/browse/track/73204121?u
<img width="581" alt="Screenshot 2024-09-23 alle 15 25 55" src="https://github.com/user-attachments/assets/469fa862-697d-44ae-baf8-b414360daf48">

https://tidal.com/browse/track/278107180?u
<img width="581" alt="Screenshot 2024-09-23 alle 15 26 06" src="https://github.com/user-attachments/assets/793f7b90-4788-4beb-a73e-61b84851cb69">

https://tidal.com/browse/track/278107187?u
<img width="581" alt="Screenshot 2024-09-23 alle 15 26 19" src="https://github.com/user-attachments/assets/59bd1762-2789-4769-b2ed-12aba9f92e86">

https://tidal.com/browse/track/75413013?u
<img width="581" alt="Screenshot 2024-09-23 alle 15 26 28" src="https://github.com/user-attachments/assets/d24973ef-69d9-4421-b396-f11b7c201c83">

https://tidal.com/browse/track/201858629?u
<img width="581" alt="Screenshot 2024-09-23 alle 15 26 40" src="https://github.com/user-attachments/assets/1bf64dac-3d99-4920-9bcd-3cd3fefe046b">

https://tidal.com/browse/track/77610758?u
<img width="581" alt="Screenshot 2024-09-23 alle 15 26 59" src="https://github.com/user-attachments/assets/782c7e57-b55d-438b-9e35-a7f5e3f08717">

https://tidal.com/browse/track/2211781?u
<img width="581" alt="Screenshot 2024-09-23 alle 15 27 08" src="https://github.com/user-attachments/assets/ebca6589-350d-4669-b69f-d7921167b828">

https://tidal.com/browse/track/17825556?u
<img width="581" alt="Screenshot 2024-09-23 alle 15 27 28" src="https://github.com/user-attachments/assets/0f4242c6-8bc2-489b-aded-09c9c718e6fa">

https://tidal.com/browse/track/152246351?u
<img width="581" alt="Screenshot 2024-09-23 alle 15 27 34" src="https://github.com/user-attachments/assets/bd5c72ff-9405-4bac-bae5-2ae36b504bb8">

https://tidal.com/browse/track/109485855?u
<img width="581" alt="Screenshot 2024-09-23 alle 15 27 54" src="https://github.com/user-attachments/assets/0ed378a9-8cc3-42a8-ad2a-2e3151a815cf">

Some are very strange and surprising. Thanks btw for all the work, if I could be helpful I'm here, keep it up!

**Comment by Inrixia** [comment]
Awesome thanks so much. These examples will be really helpful! ❤️

**Comment by Inrixia** [comment]
This should be resolved now. 

I had a whole thing with properly handling various unexpected disambiguation's but then I realized that Tidal is afik actually quite good at having the Album titles being correct and its mostly the track titles that benefit from using MusicBrainz metadata.

That said, this does mean that if Tidal gets a album title wrong or its missing its disambiguation it will be incorrect. Eg tidal has a album as "Album name" and not "Album name (Deluxe)". 

I don't think there is any scenario I have found so far where this is a issue for album names, compared to track names where it can be quite frequent.
